### DISCUSSION TITLE
Help setting up login through GDM without systemd

### DISCUSSION THREAD

**Discussion by ChrisW-B** [YaLTeR/niri#644] [Q&A]
Hey, I'm running Niri on Void Linux (so runit instead of systemd is the main concern). I had everything working, and then I decided to mess with the config without backing it up so now... it doesn't work. I can still launch niri from a TTY and have everything running as expected, but I can't seem to get GDM to launch niri.

Here's my current setup:

 I've been building directly from the repo with cargo with a script to auto copy the built files over, the script is

```shell
cargo build --release --no-default-features --features dbus,xdp-gnome-screencast
sudo cp --verbose target/release/niri /usr/bin/
sudo cp --verbose resources/niri-session /usr/bin/
# sudo cp --verbose resources/niri.desktop /usr/share/wayland-sessions/
sudo cp --verbose resources/niri-portals.conf /usr/share/xdg-desktop-portal/
```

The relevant services I have starting with the system are
- alsa
- avahi-daemon
- chronyd
- crond
- dbus
- gdm

My `/usr/share/wayland-sessions/niri.desktop` file, which gdm uses is

```desktop
[Desktop Entry]
Name=Niri
Comment=A scrollable-tiling Wayland compositor
Exec=dbus-run-session niri --session
Type=Application
DesktopNames=niri
```

Again, running `dbus-run-session niri --session` from a TTY works perfectly, so I think the problem here is more related to GDM (so advice on debugging gdm launches might also be helpful!)

Thanks!

**Comment by YaLTeR** [comment]
Weird, that sounds like it should work. Do you still need dbus-run-session in the .desktop file even when running from GDM?

**Reply by ChrisW-B** [reply]
Nope, no luck with that unfortunately. My best guess now is that there's some environment variable that I'm missing when starting from GDM that's there after logging in to TTY, maybe
